src/v/cluster/cluster_utils.h

```diff
@@ -213,25 +213,35 @@ ss::future<std::error_code> replicate_and_wait(
   Cmd&& cmd,
   model::timeout_clock::time_point timeout,
   std::optional<model::term_id> term = std::nullopt) {
-    const bool use_serde_serialization = feature_table.local().is_active(
```

Comment:
what if there is inconsistency between the boolean feature manager returns and the static type? Highly unlikely but maybe an assert or static_assert (would be even better) would be nice to provide a simple sanity check